packages/stable/src/api/dataPoints/dataPointsApi.ts

```diff
@@ -50,6 +52,65 @@ export class DataPointsAPI extends BaseResourceAPI<
     return this.retrieveDatapointsEndpoint(query);
   };
 
+  /**
+   * [Retrieve data points](https://doc.cognitedata.com/api/v1/#operation/getMultiTimeSeriesDatapoints)
+   *
+   * ```js
+   * const data = await client.datapoints.retrieveMonthlyGranularity({ items: [{ id: 123 }] });
+   * ```
+   */
+  public retrieveMonthlyGranularity = async (
+    query: DatapointsMonthlyGranularityMultiQuery
+  ): Promise<DatapointAggregates[] | Datapoints[]> => {
+    // Find the start and end dates from the query
+    const startDate = query.start;
+    const endDate = query.end;
+
+    if (startDate && endDate) {
+      // Get the months between the start and end dates
+      const months = getMonthsBetweenDates(startDate, endDate);
+
+      // Create a array of promises for each month
+      const promises = months.map((month) => {
+        // Create a new query for each month
+        const newQuery = {
+          ...query,
+          start: month.startDate,
+          end: month.endDate,
+          granularity: `${month.numberOfDays}d`,
+        };
+
+        // Return a promise for each month
+        return this.retrieveDatapointsEndpoint(newQuery);
+      });
+
+      // Call the API for each month in parallel and save it in a variable
+      // eslint-disable-next-line @typescript-eslint/no-unused-vars
+      const results = await Promise.all(promises);
+
+      // Merge the datapoints into a single array
+      const mergedDatapoints = [];
+
+      for (const result of results) {
+        // Check if the array contains data before pushing it to the merged datapoints array
+        if (Array.isArray(result) && result.length === 1) {
```

Comment:
Is there a case when result isn't an array?

Reply:
good point, it will always be one of these `DatapointAggregates[] | Datapoints[]` it could be just `result?.length`